apps/demo/src/app/cart/components/cart/cart.component.spec.ts

```diff
@@ -1,75 +1,196 @@
   });
 
-  it('renders a cart', () => {
-    expect(fixture.debugElement.query(By.css('.cart'))).not.toBeNull();
+  it('should be able to take cart as input', () => {
+    expect(component.cart).toEqual(cart);
   });
 
-  it('can be passed a Cart object', () => {
-    expect(cart.cart).toEqual(mockCart);
+  describe('on <demo-cart>', () => {
+    it('should pass down the cart', () => {
+      expect(cartItemsElement.componentInstance.cart).toEqual(cart);
+    });
   });
 
-  it('renders a <demo-cart-item> for every cart.items', () => {
-    expect(cartItems.length).toEqual(mockCart.items.length);
+  describe('on <demo-cart-item-count>', () => {
+    it('should set itemCount to value grabbed from store during init', () => {
+      cartItemCountSelector.setResult(15);
+      store.setState({});
+      fixture.detectChanges();
+
+      expect(cartItemCountElement.componentInstance.itemCount).toEqual(15);
+    });
+  });
+
+  describe('when CartContainer.$loading is false', () => {
+
+    it('should render <demo-cart>', () => {
+      expect(cartItemsElement).not.toBeNull();
+    });
+
+    it('should render <demo-help-box>', () => {
+      expect(helpBoxElement).not.toBeNull();
+    });
+
+    it('should render <demo-cart-item-count>', () => {
+      expect(cartItemCountElement).not.toBeNull();
+    });
+
+    describe('and cart is empty', () => {
+
+      beforeEach(() => {
+        cartEmptySelector.setResult(true);
+        store.setState({});
+        fixture.detectChanges();
+
+        cartSidebarElement = fixture.debugElement.query(By.css('demo-cart-sidebar'));
+      });
+
+      it('should not render <demo-cart-sidebar>', () => {
+        expect(cartSidebarElement).toBeNull();
+      });
+    });
+
+    describe('and cart is not empty', () => {
+
+      beforeEach(() => {
+        cartEmptySelector.setResult(false);
+        store.setState({});
+        fixture.detectChanges();
+
+        cartSidebarElement = fixture.debugElement.query(By.css('demo-cart-sidebar'));
+      });
+
+      it('should render <demo-cart-sidebar>', () => {
+        expect(cartSidebarElement).not.toBeNull();
+      });
+
+      describe('on <demo-cart-sidebar>', () => {
+
+        it('should pass down the cart', () => {
+          expect(cartSidebarElement.componentInstance.cart).toEqual(cart);
+        });
+      });
+    });
+  });
+
+  describe('isCartEmpty$', () => {
+    it('returns cartSelector.isCartEmpty', () => {
+      const expected = cold('(a)', { a: stubIsCartEmpty });
+      expect(component.isCartEmpty$).toBeObservable(expected);
+    });
   });
 
-  describe('on <demo-cart-item>', () => {
-    
-    it('should set item', () => {
-      expect(cartItems[0].componentInstance.item).toEqual(mockCart.items[0]);  
+  describe('itemCount$', () => {
+    it('returns cartSelector.itemCount', () => {
+      const expected = cold('(a)', { a: stubSelectCartItemCount });
+      expect(component.itemCount$).toBeObservable(expected);
     });
   });
```

Comment:
lines 183-195 should be deleted. Tests that were there before and don't do anything.